sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,278 @@ mod test_kubernetes_bootstrap_token {
+        Ok(ReservedResourcesKey {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(ReservedResourcesKey, "ReservedResourcesKey");
+
+#[cfg(test)]
+mod test_reserved_resources_key {
+    use super::ReservedResourcesKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_reserved_resources_key() {
+        for ok in &["cpu", "memory", "ephemeral-storage"] {
+            ReservedResourcesKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_reserved_resources_key() {
+        for err in &["", "cpa", ".bad", "bad.", &"a".repeat(64)] {
+            ReservedResourcesKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// ReservedResourcesValue represents a string that contains a valid Kubernetes kubeReserved
+/// and systemReserved resources value i.e. 70m, 1024Mi.
+/// https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ReservedResourcesValue {
+    inner: String,
+}
+
+impl TryFrom<&str> for ReservedResourcesValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+
+        ensure!(
+            !input.is_empty(),
+            error::BigPattern {
+                thing: " Reserved sources value",
+                input
+            }
+        );
+
+        ensure!(
+            input.chars().all(char::is_numeric) ||
+            input.ends_with("m") || input.ends_with("Mi") || input.ends_with("Gi"),
+            error::BigPattern {
+                thing: "Reserved sources value",
+                input
+            }
+        );
+
+        Ok(ReservedResourcesValue {
+            inner: input.to_string(),
+        })
+    }
+}
+string_impls_for!(ReservedResourcesValue, "ReservedResourcesValue");
+
+#[cfg(test)]
+mod test_reserved_resources_value {
+    use super::ReservedResourcesValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn good_reserved_resources_value() {
+        for ok in &["100", "100m", "1Gi", "100Mi"] {
+            ReservedResourcesValue::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn bad_reserved_resources_value() {
+        for err in &["", ".bad", "bad.", &"a".repeat(64)] {
+            ReservedResourcesValue::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// CpuManagerPolicy represents a string that contains a valid cpu management policy. Default: None
+/// https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CpuManagerPolicy {
+    inner: String,
+}
+
+impl TryFrom<&str> for CpuManagerPolicy {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        ensure!{
+            matches!(input, "static"),
```

Comment:
According to [doc](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#configuration) it supports only two values `None` and `Static`. We can use enum here too ? 

This setting also looks like `Option` : `None` or`Some('Static')` but I am not able to wrap my head around, how it will be represented to parse with serde. Serde does support something like 
```
 enum {
     CpuManagerPolicy : Option<bool>,
} 
```
But this is works when you want to read `CpuManagerPolicy` as None when value setting is not set instead of parsing its value as None. May be others would have some suggestion here.

Reply:
Yeah. I think I have a idea that set default 'None' for 'cpuManagerPolicy' if customers do not specify in userdata. like what we do for 'maxPods'. maybe that's better.